### PR TITLE
Terraform documentation

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

### <samp>🤖 Generated by Copilot at 65c4a0d</samp>

Add a new document with guidelines and best practices for Terraform config in test-infra repository. The document covers the standard Terraform configuration that is used in kyma-project/test-infra repository.

Changes proposed in this pull request:

### <samp>🤖 Generated by Copilot at 65c4a0d</samp>

*  Add a new document `docs/how-to/how-to-create-standard-terraform-config.md` that describes the standard Terraform configuration in test-infra repository ([link](https://github.com/kyma-project/test-infra/pull/7570/files?diff=unified&w=0#diff-5146126483599261f5475d42abd5c9d01fdbb639f2e5e581c835aa3002cf72acR1-R74))

**Related issue(s)**
Resolves: #7542

### CODE REVIEW

File: docs/how-to/how-to-create-standard-terraform-config.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+### Terraform presubmit prowjob
+
+Example of the [presubmit prowjob](https://github.com/kyma-project/test-infra/blob/4540c0ba3622b4f1fed47a50dedc189fdfc324b1/prow/jobs/test-infra/secrets-rotator.yaml#L92) for the secrets-rotator application. Presubmit runs terraform plan command and publish results on a GitHub pull request.
+
+Example of the [postubmit prowjob](https://github.com/kyma-project/test-infra/blob/4540c0ba3622b4f1fed47a50dedc189fdfc324b1/prow/jobs/test-infra/secrets-rotator.yaml#L222) for the secrets-rotator application. A Postsubmit runs terraform apply command and publish results on a GitHub pull request.
```
Comment (NHingerl) [inline docs/how-to/how-to-create-standard-terraform-config.md:74]:
```suggestion
Example of the [postubmit ProwJob](https://github.com/kyma-project/test-infra/blob/4540c0ba3622b4f1fed47a50dedc189fdfc324b1/prow/jobs/test-infra/secrets-rotator.yaml#L222) for the secrets-rotator application. The postsubmit ProwJob runs the `terraform apply` command and publishes the results on a GitHub pull request.
```

**Review by NHingerl — COMMENTED** [review]
excellently written. Don't worry about the number of suggestions, many of them just add a "the" here and there :D

File: docs/how-to/how-to-create-standard-terraform-config.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+#### Environment directory structure
+
+- `main.tf` file contains definition of resources that are created by the environment. Resources definition can be split into multiple files. Calls to the modules should be defined in the these files. Outputs returned by the environments can be defined in these files instead of `output.tf` file.
+- `variables.tf` file contains definition of variables that are passed to the environment and used for applying the environment. Variables defined in `variables.tf` file define environment external API. Variables defined in `variables.tf` file must be documented using the description attribute. Values of the variables defined in `variables.tf` file should be provided in the `terraform.tfvars` file.
+- `backend.tf` file contains definition of the backend that is used by the module. Terraform modules must use Google gcp as a remote state storage.
+- `terraform.tfvars` file is used to define the values of the variables that are used by the environment. The `terraform.tfvars` file should be created for each environment and should be stored in the environment directory. The path to the `terraform.tfvars` file is passed to the terraform cli command.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply docs/how-to/how-to-create-standard-terraform-config.md:63]: The `terraform` is a command which is written lower cases. Should I use backtiks?
```suggestion
- `terraform.tfvars` file is used to define the values of the variables that are used by the environment. The `terraform.tfvars` file should be created for each environment and should be stored in the environment directory. The path to the `terraform.tfvars` file is passed to the `terraform` cli command.
```

[the remark this answers is not in the document]
Comment (NHingerl) [reply docs/how-to/how-to-create-standard-terraform-config.md:82]: Yes, [commands in backticks](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md) :)

**Comment by dekiel** [comment]
@NHingerl all corrections in place. Please review.

**Pull request merged**